File: book-service/src/main/java/com/books/book_service/dto/BookResponse.java
package com.books.book_service.dto;

import com.books.book_service.model.Book;
import com.books.book_service.model.enums.BookGenres;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.NonNull;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record BookResponse(
    UUID id,
    UUID authorId,
    String title,
    String description,
    List<String> genres,
    LocalDateTime publishDate
) implements Serializable {
    public static BookResponse toDto(Book book) {
        return new BookResponse(
                book.getId(),
                book.getAuthorId(),
                book.getTitle(),
                book.getDescription(),
                List.of(book.getGenres().toString()),
                book.getPublishDate()
        );
    }
}

File: review-service/src/main/kotlin/com/books/review_service/dto/BookResponse.kt
package com.books.review_service.dto

import java.time.LocalDate
import java.util.UUID

data class BookResponse(
    val id: UUID,
    val title: String,
    val authorId: String,
    val description: String,
    val genres: List<String>,
    val publishedDate: LocalDate?
)
